doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -924,6 +924,15 @@ notations are given below. The optional :production:`scope` is described in
           given to some notation, say ``"{ y } & { z }"`` in fact applies to the
           underlying ``"{ x }"``\-free rule which is ``"y & z"``).
 
+.. note:: Notations such as ``"( p | q )"`` (or starting with ``"( x | "``,
+          more generally) are deprecated as they conflict with the syntax for
+          nested disjunctive patterns (see :ref:`extendedpatternmatching`),
+          and are not honored in pattern expressions.
+
+.. warn:: Use of @string Notation is deprecated as it is inconsistent with pattern syntax.
+
+          The ``-w disj-pattern-notation`` flag turns on this warning.
```

Comment:
Even if it is implicit, I would mention that it is off by default (and the reason). I would also indent the `.. warn::` directive within the note body. Note that you are allowed to indent the documentation of the warning by only three characters (i.e. aligning it with `warn`).

Reply:
Is the latest revision what you had in mind? Do note that indenting the `..warn` makes the warning message stick out the right margin.
  I've given minimal rationale, so as not to single out `Numbers`. It would make sense to revise that library's use of an `(p | q)` notation once PR #10180 is merged, and perhaps turn on this warning by default if it can be disabled for deprecated notation (changing the reman paragraph yet again!).